// bmkg-wrapper/dist/cjs/api/gempa/autoGempa.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.autoGempa = void 0;
const axios_1 = __importDefault(require("axios"));
const constants_js_1 = require("../../config/constants.js");
/**
 * Mengambil data gempa yang baru-baru ini terjadi.
 * @returns Object berisi data gempa yang baru-baru ini terjadi.
 */
const autoGempa = async () => {
    var _a;
    try {
        const res = await axios_1.default.get(constants_js_1.ENDPOINTS.GEMPA.AUTO_GEMPA);
        const gempa = (_a = res.data.Infogempa) === null || _a === void 0 ? void 0 : _a.gempa;
        if (!gempa)
            return null;
        return {
            tanggal: gempa.Tanggal,
            jam: gempa.Jam,
            dateTime: gempa.DateTime,
            coordinates: gempa.Coordinates,
            lintang: gempa.Lintang,
            bujur: gempa.Bujur,
            magnitude: Number(gempa.Magnitude),
            kedalaman: gempa.Kedalaman,
            wilayah: gempa.Wilayah,
            potensi: gempa.Potensi,
            dirasakan: gempa.Dirasakan,
            shakemap: constants_js_1.BASE_URL.GEMPA + '/' + gempa.Shakemap,
        };
    }
    catch (err) {
        console.error('Error fetching data gempa:', err);
        return null;
    }
};
exports.autoGempa = autoGempa;
