// bmkg-wrapper/dist/cjs/api/gempa/gempaDirasakan.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.gempaDirasakan = void 0;
const axios_1 = __importDefault(require("axios"));
const constants_js_1 = require("../../config/constants.js");
/**
 * Mengambil list data gempa yang dirasakan.
 * @returns Array berisi data gempa yang dirasakan.
 */
const gempaDirasakan = async () => {
    var _a, _b;
    try {
        const res = await axios_1.default.get(constants_js_1.ENDPOINTS.GEMPA.GEMPA_DIRASAKAN);
        const gempa = (_b = (_a = res.data.Infogempa) === null || _a === void 0 ? void 0 : _a.gempa) !== null && _b !== void 0 ? _b : [];
        return gempa.map(({ Tanggal, Jam, DateTime, Coordinates, Lintang, Bujur, Magnitude, Kedalaman, Wilayah, Dirasakan, }) => ({
            tanggal: Tanggal,
            jam: Jam,
            dateTime: DateTime,
            coordinates: Coordinates,
            lintang: Lintang,
            bujur: Bujur,
            magnitude: Number(Magnitude),
            kedalaman: Kedalaman,
            wilayah: Wilayah,
            dirasakan: Dirasakan,
        }));
    }
    catch (err) {
        console.error('Error fetching data gempa:', err);
        return [];
    }
};
exports.gempaDirasakan = gempaDirasakan;
